test/token-verification.test.js

```diff
@@ -118,113 +157,448 @@ describe('jwt-verification', function() {
+                token,
+                done
+              );
+            })
+            .catch(done);
+        });
+
+        it('should be valid when auth_time is within the leeway', done => {
+          const now = new Date();
+          const maxAge = 1000;
+          const nowSeconds = Math.floor(now.getTime() / 1000) - 10;
+          const leeway = 20;
+          const validUntil = nowSeconds + maxAge + leeway;
+          const validUntilDate = new Date(0);
+
+          validUntilDate.setUTCSeconds(validUntil);
+
+          const config = Object.assign({}, DEFAULT_CONFIG, {
+            maxAge,
+            leeway
+          });
+
+          const payload = Object.assign({}, DEFAULT_PAYLOAD, {
+            auth_time: nowSeconds
+          });
+
+          createJWT(payload, DEFAULT_OPTIONS)
+            .then(token => {
+              new IdTokenVerifier(config).verify(
+                token,
+                'asfd',
+                (err, result) => {
+                  expect(err).to.be(null);
+                  done();
+                }
+              );
+            })
+            .catch(done);
         });
       });
     });
-    describe('without stubing `getRsaVerifier`', () => {
+
+    describe('without stubbing `getRsaVerifier`', () => {
       beforeEach(() => {
         global.fetch = nodeFetch;
       });
+
       afterEach(() => {
         global.fetch = undefined;
       });
+
       it('should fail with corrupt token', done => {
         helpers.assertTokenValidationError(
-          {
-            issuer: 'https://wptest.auth0.com/',
-            audience: 'gYSNlU4YC4V1YPdqq8zPQcup6rJw1Mbt'
-          },
+          DEFAULT_CONFIG,
           'asfd',
           'Invalid signature.',
           'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IlF6RTROMFpCTTBWRFF6RTJSVVUwTnpJMVF6WTFNelE0UVRrMU16QXdNRUk0UkRneE56RTRSZyJ9.eyJpc3MiOiJodHRwczovL3dwdGVzdC5hdXRoMC5jb20vIiwic3ViIjoiYXV0aDB8NTVkNDhjNTdkNWIwYWQwMjIzYzQwOGQ3IiwiYXVkIjoiZ1lTTmxVNFlDNFYxWVBkcXE4elBRY3VwNnJKdzFNYnQiLCJleHAiOjk0ODI5NjkwMzEsImlhdCI6MTQ4MjkzMzAzMSwibm9uY2UiOiJhc2ZkIn0.PPoh-pITcZ8qbF5l5rMZwXiwk5efbESuqZ0IfMUcamB6jdgLwTxq-HpOT_x5q6-sO1PBHchpSo1WHeDYMlRrOFd9bh741sUuBuXdPQZ3Zb0i2sNOAC2RFB1E11mZn7uNvVPGdPTg-Y5xppz30GSXoOJLbeBszfrVDCmPhpHKGGMPL1N6HV-3EEF77L34YNAi2JQ-b70nFK_dnYmmv0cYTGUxtGTHkl64UEDLi3u7bV-kbGky3iOOCzXKzDDY6BBKpCRTc2KlbrkO2A2PuDn27WVv1QCNEFHvJN7HxiDDzXOsaUmjrQ3sfrHhzD7S9BcCRkekRfD9g95SKD5J0Fj8NA',
           done
         );
       });
+
       it('should fetch the public key and verify the token ', done => {
         helpers.assertTokenValid(
+          defaultToken,
           {
             issuer: 'https://wptest.auth0.com/',
             audience: 'gYSNlU4YC4V1YPdqq8zPQcup6rJw1Mbt',
-            __disableExpirationCheck: true
+            __disableExpirationCheck: true,
+            __clock: () => new Date('Wed Dec 28 2016 23:00:00 GMT+0000')
```

Comment:
you might want to move this up into a constant. I see you're using it on other lines as well